extensions/resteasy-reactive/quarkus-resteasy-reactive-common/runtime/src/main/java/io/quarkus/resteasy/reactive/common/runtime/JaxRsSecurityConfig.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.resteasy.reactive.common.runtime;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+import io.smallrye.config.WithName;
+
+@ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+@ConfigMapping(prefix = "quarkus.security.jaxrs")
+public interface JaxRsSecurityConfig {
```

Comment:
For some time now. The support was added in https://github.com/quarkusio/quarkus/pull/19230.

There shouldn't be any downsides. We must combine both annotations in the mapping class to state the build phase. Eventually, we should just have an annotation for that, but since this was still experimental, I didn't want to add yet another API. Also, a lot of our support code for configuration extension expects to find a `@ConfigRoot` annotation, so it was easier this way.

The only thing I remember I didn't implement is the injection of method parameters in recorder classes. It required some extra work, and I didn't want to delay the original work further. Using a setter from the Processor to the Recorder does work.

Of course, there may still be bugs in there and I'll fix them when we find them :)

Reply:
Great to know, thanks!